internal/daemon/controller/handler.go

```diff
@@ -737,3 +738,20 @@ func wrapHandlerWithCallbackInterceptor(h http.Handler, c *Controller) http.Hand
 		h.ServeHTTP(w, req)
 	})
 }
+
+// getActions takes in a URL Path and returns the actions from the URL
+func getActions(path string) []string {
+	// Remove any query parameters
+	path = strings.Split(path, "?")[0]
+
+	parts := strings.Split(path, "/")
+	lastPart := parts[len(parts)-1]
+
+	items := strings.Split(lastPart, ":")
+
+	if len(items) > 1 {
+		return items[1:]
+	}
+
+	return []string{}
```

Comment:
Maybe a little simpler. Up to you.
```suggestion
	// Cut off everything before the first ":"
	_, rest, _ := strings.Cut(lastPart, ":")
	
	// Split the rest on ":", returning all actions and subactions
	return strings.Split(rest, ":")
```